flink-connector-aws/flink-connector-sqs/src/main/java/org/apache/flink/connector/sqs/sink/SqsSinkWriter.java

```diff
@@ -0,0 +1,222 @@
+                        .setMaxTimeInBufferMS(maxTimeInBufferMS)
+                        .setMaxRecordSizeInBytes(maxRecordSizeInBytes)
+                        .build(),
+                initialStates);
+        this.failOnError = failOnError;
+        this.sqsUrl = sqsUrl;
+        this.metrics = context.metricGroup();
+        this.numRecordsOutErrorsCounter = metrics.getNumRecordsOutErrorsCounter();
+        this.clientProvider = clientProvider;
+    }
+
+    @Override
+    protected void submitRequestEntries(
+            List<SendMessageBatchRequestEntry> requestEntries,
+            Consumer<List<SendMessageBatchRequestEntry>> requestResult) {
+
+        final SendMessageBatchRequest batchRequest =
+                SendMessageBatchRequest.builder().entries(requestEntries).queueUrl(sqsUrl).build();
+
+        CompletableFuture<SendMessageBatchResponse> future =
+                clientProvider.getClient().sendMessageBatch(batchRequest);
+
+        future.whenComplete(
+                (response, err) -> {
+                    if (err != null) {
+                        handleFullyFailedRequest(err, requestEntries, requestResult);
+                    } else if (response.failed() != null && response.failed().size() > 0) {
+                        handlePartiallyFailedRequest(response, requestEntries, requestResult);
+                    } else {
+                        requestResult.accept(Collections.emptyList());
+                    }
+                });
+    }
+
+    @Override
+    protected long getSizeInBytes(SendMessageBatchRequestEntry requestEntry) {
+        return requestEntry.messageBody().getBytes(StandardCharsets.UTF_8).length;
+    }
+
+    @Override
+    public void close() {
+        AWSGeneralUtil.closeResources(clientProvider);
+    }
+
+    private void handleFullyFailedRequest(
+            Throwable err,
+            List<SendMessageBatchRequestEntry> requestEntries,
+            Consumer<List<SendMessageBatchRequestEntry>> requestResult) {
+
+        numRecordsOutErrorsCounter.inc(requestEntries.size());
+        boolean isFatal = SQS_EXCEPTION_HANDLER.consumeIfFatal(err, getFatalExceptionCons());
+        if (isFatal) {
+            return;
+        }
+
+        if (failOnError) {
+            getFatalExceptionCons().accept(new SqsSinkException.SqsFailFastSinkException(err));
+            return;
+        }
+
+        LOG.warn(
+                "SQS Sink failed to write and will retry {} entries to SQS first request was {}",
+                requestEntries.size(),
+                requestEntries.get(0).toString(),
+                err);
+        requestResult.accept(requestEntries);
+    }
+
+    private void handlePartiallyFailedRequest(
+            SendMessageBatchResponse response,
+            List<SendMessageBatchRequestEntry> requestEntries,
+            Consumer<List<SendMessageBatchRequestEntry>> requestResult) {
+
+        if (response.failed() != null) {
```

Comment:
This `if` is redundant as `handlePartiallyFailedRequest` is only called when `response.failed()!=null)`

Reply:
Good catch, Removed